internal/pkg/template/diff/list.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+func longestCommonSubsequence[T comparable](a []T, b []T) []T {
```

Comment:
How are we going to use this with `yaml.Node`s? 😅 

Do you think that having a signature like this makes sense:
```suggestion
// Returns the longest common sequence between a and b, as well as the indices where each sequence items appeared in a and b.
func longestCommonSubsequence[T comparable](a []T, b []T) ([]T, []int, []int) {
```

My thought was that we can pass in the string representation of each `yaml.Node` in `a` and `b` and since we get the indices we would have a way of telling if the cursor should be deleted or added

Reply:
> My thought was that we can pass in the string representation of each yaml.Node in a and b

Change of mind! This is not the best way to compare two list items. Consider this:
```
- Sid: power
  Action: '*'
  Resources: '*'

# Compare with

- Sid: power
  Resources: '*'
  Action: '*'
```

The only thing that changed as the order of the properties, but it will produce a diff. CloudFormation does not create a change set in this case either. 

My plan now is to something like:
```go
// var fromSeq, toSeq []yaml.Node
lcsInFrom, lcsInTo := longestCommonSubsequence(fromSeq, toSeq, func(inA, inB int) bool {
   diff, err := parse(fromSeq[inA], toSeq[inB], "") // A recursive call to "parse". 
   if err != nil {
      return false
   }
   return diff == nil
})

```